Update Adding Functionality of Keyword Generations

positiveVibes4all opened the pull request:
## Summary 
Changed the source of the tags from context to the tags property so that the UI does not wait for the counts to come back to re-render. Refactored the component to its own file to help clean up the large file for Expandable tags. 

### List of updates
- Allows for the user to click as many keywords as desired while the counts load, some pages are much slower for counts to come back on and this fixes that.
- Refactor code to clean up the component file in regard to keyword generations.
- Fix height of keyword generation tag buttons

decrapifier commented:
<!--
  2 failure:  The changelog has..., Please add a desc...
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>The changelog has not been updated. Please update the changelog.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Please add a description to your PR</td>
    </tr>
  </tbody>
</table>



## Coverage Report

> :+1: Test coverage is looking good.

|Files|% Stmts|% Branch|% Funcs|% Lines|Uncovered Lines||
|---|:-:|:-:|:-:|:-:|:-:|:-:|
|[src/dataStrategies.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js)|85|100|83.33|85|[28](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L28), [29](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L29), [30](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L30), [31](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L31), [32](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L32), [33](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L33), [34](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L34), [35](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L35), [36](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/dataStrategies.js#L36)|:white_check_mark:|
|[src/exampleTypes.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/exampleTypes.js)|87.76|80|100|87.76|[10](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/exampleTypes.js#L10), [11](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/exampleTypes.js#L11), [12](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/exampleTypes.js#L12), [13](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/exampleTypes.js#L13), [14](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/exampleTypes.js#L14), [15](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/exampleTypes.js#L15)|:white_check_mark:|
|[src/index.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/index.js)|94.74|93.33|100|94.74|[48](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/index.js#L48), [49](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/index.js#L49), [58](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/index.js#L58), [59](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/index.js#L59)|:white_check_mark:|

<details>
<summary>
and 7 more...
</summary>

|Files|% Stmts|% Branch|% Funcs|% Lines|Uncovered Lines||
|---|:-:|:-:|:-:|:-:|:-:|:-:|
|[src/utils.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js)|84.95|94.59|100|84.95|[44](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L44), [45](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L45), [46](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L46), [47](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L47), [48](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L48), [49](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L49), [64](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L64), [65](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L65), [66](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L66), [67](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/utils.js#L67)...|:white_check_mark:|
|[src/__data__/imdb.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/__data__/imdb.js)|100|100|100|100||:white_check_mark:|
|[src/provider-debug/index.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-debug/index.js)|100|100|100|100||:white_check_mark:|
|[src/provider-memory/date.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/date.js)|94.55|94.44|60|94.55|[9](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/date.js#L9), [10](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/date.js#L10), [11](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/date.js#L11), [64](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/date.js#L64), [65](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/date.js#L65), [66](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/date.js#L66)|:white_check_mark:|
|[src/provider-memory/<br>exampleTypes.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js)|91.26|92|82.35|91.26|[16](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L16), [17](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L17), [18](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L18), [96](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L96), [97](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L97), [98](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L98), [99](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L99), [100](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L100), [101](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/exampleTypes.js#L101)|:white_check_mark:|
|[src/provider-memory/<br>index.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/index.js)|100|75|100|100||:white_check_mark:|
|[src/provider-memory/<br>results.js](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/results.js)|94.44|66.67|100|94.44|[14](../blob/028779147508a8f5bb5a3d54d05137fdea948304/src/provider-memory/results.js#L14)|:white_check_mark:|
</details>
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 028779147508a8f5bb5a3d54d05137fdea948304
</p>

lint-action committed:
Fix code style issues with Prettier

Commit:
Fix Loading Between Keyword Adds

- Fixed loading between keyword additions so that multiple can be added without disrupting the flow

Commit:
Update Typo

Commit:
Update Height of Tags

- Updated keyword height to match tags.

lint-action committed:
Fix code style issues with Prettier

Commit:
BUMP

Commit:
Merge remote-tracking branch 'origin/feature/KEYWORD-GENERATIONS-UPDATE-BUT-DONT-REMOVE-KEYWORDS' into feature/KEYWORD-GENERATIONS-UPDATE-BUT-DONT-REMOVE-KEYWORDS

Commit:
Merge branch 'main' into feature/KEYWORD-GENERATIONS-UPDATE-BUT-DONT-REMOVE-KEYWORDS

Commit:
PR Feedback

lint-action committed:
Fix code style issues with Prettier

re1ro commented on packages/react/src/exampleTypes/ExpandableTagsQuery/index.js:
```diff
@@ -166,10 +109,15 @@ let TagsWrapper = observer(
   }) => {
+    //Handle Outside Clicks for Keyword Generations display
+    let ref = React.useRef()
+    let generationsCollapsed = React.useState(true)
```
We already have `generationsCollapsed` passed as a prop to this component, why do we need to define another one? 

If we absolutely need to define another one you need to check is there is popup open in the outside click handler like we do here

https://github.com/smartprocure/contexture/blob/349793ca2b3459ae25c5185870bf90a8904c399d/packages/react/src/exampleTypes/ExpandableTagsQuery/index.js#L60

stellarhoof replied:
@positiveVibes4all The entire outside click handler + `generationsCollapsed` prop + ref are not being used

positiveVibes4all replied:
I updated the code appropriately

Commit:
PR Feedback & Other Fixes

- Feedback related to duplicated ref addressed
- Fixed bug that was preventing the keywordgenerations from collapsing
- Fixed counts bug related to making them lowercase for dedup purposes in a past commit.

Commit:
BUMP

Commit:
Margin Update

- Shows proper margin now only based on the presence of the View More Keywords button.

Commit:
Merge branch 'main' into feature/KEYWORD-GENERATIONS-UPDATE-BUT-DONT-REMOVE-KEYWORDS

lint-action committed:
Fix code style issues with Prettier
